manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1062,454 @@ def text2svg(self):
+        :save_last_frame:
+
+        class MultiLanguage(Scene):
+            def construct(self):
+                morning = MarkupText("வணக்கம்", font="sans-serif")
+                chin = MarkupText(
+                    '見 角 言 谷  辛 <color col="BLUE">辰 辵 邑</color> 酉 釆 里!'
+                )  # works as in ``Text``.
+                mess = MarkupText("Multi-Language", style=BOLD)
+                russ = MarkupText("Здравствуйте मस नम म ", font="sans-serif")
+                hin = MarkupText("नमस्ते", font="sans-serif")
+                japanese = MarkupText("臂猿「黛比」帶著孩子", font="sans-serif")
+                group = VGroup(morning, chin, mess, russ, hin, japanese).arrange(DOWN)
+                self.add(group)
+
+
+    Tests
+    -----
+
+    Check that the creation of :class:`~.MarkupText` works::
+
+        >>> MarkupText('The horse does not eat cucumber salad.')
+        MarkupText('The horse does not eat cucumber salad.')
+
+    .. WARNING::
+
+        Using a :class:`.Transform` on text with leading whitespace can look
+        `weird <https://github.com/3b1b/manim/issues/1067>`_. Consider using
+        :meth:`remove_invisible_chars` to resolve this issue.
+
+    """
+
+    def __init__(
+        self,
+        text: str,
+        fill_opacity: int = 1,
+        stroke_width: int = 0,
+        color: str = WHITE,
+        size: int = 1,
+        line_spacing: int = -1,
+        font: str = "",
+        slant: str = NORMAL,
+        weight: str = NORMAL,
+        gradient: tuple = None,
+        tab_width: int = 4,
+        height: int = None,
+        width: int = None,
+        should_center: bool = True,
+        unpack_groups: bool = True,
+        disable_ligatures: bool = False,
+        **kwargs,
+    ):
+        self.text = text
+        self.size = size
+        self.line_spacing = line_spacing
+        self.font = font
+        self.slant = slant
+        self.weight = weight
+        self.gradient = gradient
+        self.tab_width = tab_width
+
+        self.original_text = text
+        self.disable_ligatures = disable_ligatures
+        text_without_tabs = text
+        if text.find("\t") != -1:
+            text_without_tabs = text.replace("\t", " " * self.tab_width)
+
+        colormap = self.extract_color_tags()
+        gradientmap = self.extract_gradient_tags()
+
+        if self.line_spacing == -1:
+            self.line_spacing = self.size + self.size * 0.3
+        else:
+            self.line_spacing = self.size + self.size * self.line_spacing
+
+        file_name = self.text2svg()
+        self.remove_last_M(file_name)
+        SVGMobject.__init__(
+            self,
+            file_name,
+            color=color,
+            fill_opacity=fill_opacity,
+            stroke_width=stroke_width,
+            height=height,
+            width=width,
+            should_center=should_center,
+            unpack_groups=unpack_groups,
+            **kwargs,
+        )
+        self.chars = VGroup(*self.submobjects)
```

Comment:
Is the `disable_ligatures` method removed? Because there should be an if condition here to check for space and replace it with dots. `gen_chars` methods.

Reply:
The usage of `gen_chars`in case of `disable_ligatures=True` is not necessary here according to my tests (see the examples). That's because the whole formatting is done by Pango and does not rely on the character position; except for colors and gradients, which are done (currently have to be done) outside of Pango. The class uses its own method to guess the right index. It works fine for `disable_ligatures=True` but obviously not for `False`; that's when the `offset` parameter kicks in.